### PR TITLE
Fix fees

### PR DESCRIPTION
**Pull request by D4nte**
- Update time to mine values to some more realistic times from looking at mainnet Ethereum and Bitcoin fees.
- Did a fix in the `expiries` modules where "bob time to complete" would include "alice's time to act" when the clock started, making Bob abort when Alice would act after 50min (she is allowed 60min).
- Default to use `bitcoind estimatesmartfee` feature for nectar instead of static value.

### CODE REVIEW

**Review by tcharding — APPROVED** [review]

**Review by thomaseizinger — COMMENTED** [review]
What you did in patch https://github.com/comit-network/comit-rs/pull/3256/commits/327375adaf8ccad955db66aa55f474bf7599114a is unfortunately not quite enough :)

I was working on this as well and included your patch in which you fix it here: https://github.com/comit-network/comit-rs/pull/3261

**Comment by tcharding** [comment]
Added no-mergify because I'm half way through review and have a one-on-one call.

File: comit/src/expiries.rs [hunk 1]
```diff
@@ -712,13 +716,17 @@ impl BobState {
         };
 
         match next_action {
+            Start => {
+                // This time is needed for Alice to act.
+                // When a swap in progress, this time is not taken in
+                // account to know whether Bob can still complete.
+                c.broadcast_alpha_fund_transaction()
```
Comment (tcharding) [inline comit/src/expiries.rs:723]: This should be `c.start()` along with
```
// This time is needed for Alice to start
```

File: comit/src/expiries.rs [hunk 2]
```diff
@@ -712,13 +716,17 @@ impl BobState {
...
+            }
             // Once Alice starts we need at least this much time, note that c.start() is not
             // included otherwise the actual time Alice takes invalidates the swap for Bob.
             WaitForAlphaFundTransactionFinality => {
                 // Transition from Started to AlphaFunded
-                c.broadcast_alpha_fund_transaction()
-                    + c.mine_alpha_fund_transaction()
-                    + c.finality_alpha()
```
Comment (tcharding) [inline comit/src/expiries.rs:721]: I don't agree with this change, can you explain please? Transitioning from Started to AlphaFunded includes the broadcast_aplha_fun_transaction period.

**Review by tcharding — DISMISSED** [review]
> - Update time to mine values to some more realistic times from looking at mainnet Ethereum and Bitcoin fees.
> - Did a fix in the expiries modules where "bob time to complete" would include "alice's time to act" when the clock started, making Bob abort when Alice would act after 50min (she is allowed 60min).
> - Default to use bitcoind estimatesmartfee feature for nectar instead of static value.

ACK the first and last. NACK the second, I do not think this is a good change mate. I understand from the comment that you saw unexpected results that Bob aborted when Alice takes a while to act. I do not think that we should be making tweaks to the expiries state transition periods just to fix edge cases where actors act close to deadlines. The whole module is hueristic at best and is no way safe to push to the edge. On top of this the transition period calculations changed in this PR are incorrect IMO, commented inline.

**Comment by thomaseizinger** [comment]
FYI: https://github.com/comit-network/comit-rs/pull/3261 has been merged into `dev` already, you can now rebase and drop [`62dfe95` (#3256)](https://github.com/comit-network/comit-rs/pull/3256/commits/62dfe952d4c95c4d79a896d3bbacd5deb9d5465a).

File: comit/src/expiries.rs [hunk 3]
```diff
@@ -899,8 +908,8 @@ mod tests {
     fn dev_net_herc20_hbit_expiries() {
         let (a, b) = expiry_offsets_herc20_hbit(Network::Dev);
 
-        assert_eq!(a, 44.seconds().into());
-        assert_eq!(b, 35.seconds().into());
+        assert_eq!(a, 218.seconds().into());
+        assert_eq!(b, 155.seconds().into());
```
Comment (thomaseizinger) [inline comit/src/expiries.rs:912]: This is not good. Why did these values go up that much? If we ever want to test refund in our e2e tests, this is amount of time we would have to wait. 218 seconds is almost 4 minutes of just waiting.

Can we somehow keep these numbers under 1 minute?

Comment (D4nte) [reply comit/src/expiries.rs:912]: > This is not good. Why did these values go up that much? If we ever want to test refund in our e2e tests, this is amount of time we would have to wait. 218 seconds is almost 4 minutes of just waiting.
> 
> Can we somehow keep these numbers under 1 minute?

Yes we can with separate mainnet and dev value. Will do.

Comment (thomaseizinger) [reply comit/src/expiries.rs:912]: Cool, thank you!

File: nectar/src/bitcoin/fee.rs [hunk 4]
```diff
@@ -18,6 +18,7 @@ impl Fee {
 
     pub async fn kvbyte_rate(&self) -> Result<Amount> {
         let rate = self.vbyte_rate().await?;
+        dbg!(&rate);
```
Comment (thomaseizinger) [inline nectar/src/bitcoin/fee.rs:20]: Left-over `dbg`?

File: nectar/src/swap.rs [hunk 5]
```diff
@@ -385,7 +385,7 @@ mod tests {
                 crate::ethereum::GasPrice::geth_url(ethereum_blockchain.node_url.clone());
 
             let bitcoin_fee = crate::bitcoin::Fee::new(
-                Default::default(),
+                crate::config::BitcoinFees::SatsPerByte(bitcoin::Amount::from_sat(50)),
```
Comment (thomaseizinger) [inline nectar/src/swap.rs]: Left-over `fixup` commit :)

File: comit/src/expiries.rs [hunk 6]
```diff
@@ -4,7 +4,7 @@
 //! atomic. One expiry for the alpha ledger and one for the beta ledger. The
 //! beta ledger expiry time must not elapse before the alpha ledger expiry time.
 
-mod config;
+pub mod config;
```
Comment (thomaseizinger) [inline comit/src/expiries.rs:7]: We are re-exporting it now, this should go away :)

**Review by thomaseizinger — APPROVED** [review]
Looks good! Thank you :)

You've got a left-over `fixup` commit here: [`0e631ea` (#3256)](https://github.com/comit-network/comit-rs/pull/3256/commits/0e631ea9e756e85c1f8289cd1cff8ffbb1af69fd)

File: comit/src/expiries.rs (hunk 6 above)
Comment (D4nte) [reply comit/src/expiries.rs:7]: ooops, slipped in. thanks.

File: comit/src/actions.rs [hunk 7]
```diff
@@ -65,13 +65,10 @@ pub mod bitcoin {
     where
         C: secp256k1::Signing,
     {
-        let rate = byte_rate.as_sat();
-        // TODO: change this interface to accept an amount instead of a usize and remove
-        // this allow
         #[allow(clippy::cast_possible_truncation)]
```
Comment (thomaseizinger) [inline comit/src/actions.rs:68]: This should not longer be needed I think.
```suggestion
```

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**